### ISSUE TITLE
Should we surface a=msid changes to JS, even though they're not compliant with draft-jsep?

### ISSUE THREAD

**Issue by docfaraday** [w3c/webrtc-pc#2115]
draft-jsep says that a=msid attributes, once set, are invariant until the corresponding transceiver is stopped. webrtc-pc does not seem to say anything about this invariance, which means that if a=msid _does_ change in a remote description, we surface that change to content.

Was this invariance imposed because we did not want to surface such changes to content? Or was there some other reason?

**Comment by jan-ivar** [comment]
Same question on the other side with [sender.setStreams()](http://w3c.github.io/webrtc-pc/#dom-rtcrtpsender-setstreams) which actively changes msids.

**Comment by alvestrand** [comment]
When we defined a=msid the first time (in the context of plan B), changing the track ID (aka "appdata") was the way to say "the old track has gone away, you're now seeing a new one".

Now that tracks are identified by their tie to a media section, it's possible to make a change of appdata mean something else - but we also took away the requirement to send it, and the requirement to show it on the receiving side; we no longer require the track's "id" attribute to be derived from a=msid. So appdata is largely useless - it's not even surfaced reliably.

So that leaves the stream ID (when present). This can change, and should be interpreted as a change in the list of streams carrying the track.

My $0.02.
